vortex-sampling-compressor/tests/smoketest.rs

```diff
@@ -118,7 +118,10 @@ fn make_timestamp_column(count: usize) -> Array {
 
     let storage_array = PrimitiveArray::from_vec(timestamps, Validity::NonNullable).into_array();
 
-    TemporalArray::new_timestamp(storage_array, TimeUnit::Ms, None)
-        .into_array_data()
-        .into_array()
+    ArrayData::from(TemporalArray::new_timestamp(
```

Comment:
This could be just `Array::from`?

Reply:
True. `TemporalArray` isn't generated by `impl_encoding`, but its easy enough to add a `From implementation.